src/test/scala/com/cognite/sdk/scala/common/TokenTest.scala

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020 Cognite AS
+// SPDX-License-Identifier: Apache-2.0
+
+package com.cognite.sdk.scala.common
+
+//import java.util.concurrent.Executors
+//
+//import cats.effect.laws.util.TestContext
+//import cats.effect.{ContextShift, IO, Timer}
+//import com.cognite.sdk.scala.v1.RequestSession
+//import com.softwaremill.sttp._
+//import com.softwaremill.sttp.asynchttpclient.cats.AsyncHttpClientCatsBackend
+//
+//import scala.concurrent.ExecutionContext
+//import scala.concurrent.duration._
+
+class TokenTest extends SdkTestSpec {
```

Comment:
It looks complicated compared to LoginTest which looks like it would be similar... that one just uses `Id` instead of `IO`, and a simple `HttpURLConnectionBackend` with a timeout option.

Reply:
. I will try your suggestion, thanks!

I commented the tests out to see if they are the problem but it looks like it is still stuck.

Reply:
And these tests are successful in local :/

Reply:
Reading the test output is tricky, because if a test just gets stuck it might not print anything, so the last message you see may be from a test that did succeed.

GHA runs on tiny VMs, I think they have only two (virtual) cores, which can be a relevant difference if you accidentally end up using the global executioncontext, because that one is sized according to the number of cores. Locally you probably have at least 4 cores, probably 8 or more.

Of course, if it gets stuck even without your tests, something is up... maybe try reverting the settings in scala.yml in that case.

Reply:
As opposed to LoginTest, I need to define `val authProvider = OAuth2.ClientCredentialsProvider[Id](credentials)` and having some trouble defining implicits for it when using `Id`.

```
object ClientCredentialsProvider {
    def apply[F[_]](
        credentials: ClientCredentials,
        refreshSecondsBeforeTTL: Long = 30
    )(
        implicit F: ConcurrentEffect[F],
        clock: Clock[F],
        sttpBackend: SttpBackend[F, Nothing]
    )
```

Reply:
Ah, I see, if that one truly needs `ConcurrentEffect` we can't use `Id`, because `Id` runs each action immediately so it's not possible to provide concurrency for it. I'll clone this repo and take a closer look at what's going on, maybe I can find something.

Reply:
Thanks a lot for the help! I will revert the commented tests since it is still stuck without them. :/

Reply:
Oookay, well, `TestContext` looked weird to me, so I made this more similar to `GzipSttpBackendTest` which is just using the global execution context.

`OAuth2ClientCredentialsTest` is using `TestContext` to get a fake timer which can be driven by the test. For whatever reason that doesn't seem to work well on GitHub Actions, because since it was merged, the workflow eventually passes, but takes more than four hours: https://github.com/cognitedata/cognite-sdk-scala/actions/runs/533755064

That's not related to your change here, which looks good to me (and now, the test passes, but you need to be looking at the log output as it runs for GHA to show it to you).

Reply:
Thanks for investigating this! It now time-outs I think because of timeout-minutes in the github workflow. I will continue looking tomorrow.

Reply:
Turns out it's because we're using the Greenfield client for events, as publicdata has too many... and something was writing a lot of "SAP" (not really, I think, but that was the source) events into my tenant `wjoel` on Greenfield, which is the one we're using for `cognite-sdk-scala`, so there were >5 million events in there. That's what took hours to page through.

I couldn't figure out where they were coming from, so I revoked all keys until it stopped, but I also revoked the key used by GitHub Actions. I'll make a new one and try to get it added to the secrets.

Reply:
I seee. Then that explains why we had problems in EventsTest and GreenfieldTests.

Reply:
Yep. This is good to go! :rocket: